app/models/user.rb

```diff
@@ -1,11 +1,13 @@
 class User < ApplicationRecord
-  has_many :followers_ref, foreign_key: :follower_id , class_name: "Friendship"
+  has_many :followers_ref, foreign_key: :follower_id, class_name: 'Friendship', dependent: :destroy
   has_many :followed, through: :followers_ref
-  has_many :followed_ref, foreign_key: :followed_id, class_name: "Friendship"
+  has_many :followed_ref, foreign_key: :followed_id, class_name: 'Friendship', dependent: :destroy
   has_many :followers, through: :followed_ref
+  has_many :party_guests, dependent: :destroy
+  has_many :parties, through: :party_guests
 
   validates :email, uniqueness: true, presence: true
-  validates_presence_of :password_confirmation
+  validates :password_confirmation, presence: true
```

Comment:
Isn't this refactor saying the same thing? Curious if there is a reason to change it.